### PR TITLE
(HTCONDOR-2430)  Don't make the v2 bindings version-specific.

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
The version 2 Python bindings were written to be minor version -independent from the start, but they hadn't been being built that way.

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/python-bindings/condor_python.h [hunk 1]
```diff
@@ -35,13 +35,9 @@
...
 #if defined(_CONDOR_COMMON_FIRST)
     #include "condor_common.h"
-    // This #define confuses linking on Windows.
-    // (It links the non-minor-version-specific
-    // python3.lib, which doesn't exist.)
-    #define Py_LIMITED_API
 #endif /* _CONDOR_COMMON_FIRST */
 
-
+#define Py_LIMITED_API 3
```
Comment (GregThain) [inline src/python-bindings/condor_python.h:40]: Can we get a comment here about why "3", when we might want to change it, and what other things (like .abi3.so) that might need to change at the same time?

**Review by GregThain — CHANGES_REQUESTED** [review]
Not nearly as terrifying as I had feared.

File: src/python-bindings/condor_python.h (hunk 1 above)
Comment (Todd-L-Miller) [reply src/python-bindings/condor_python.h:40]: That's a good idea.

**Review by GregThain — APPROVED** [review]
LGTM.

**Pull request merged**